### ISSUE TITLE
Alien Swarm: Reactive Drop (563560)

### ISSUE THREAD

**Issue by josephcocoa** [ValveSoftware/Proton#942]
It looks like this is the exact same issue as what happens with the original "Alien Swarm" game (issue #276). This time though, I was able to see that it said "Node graph out of date, reloading".

It looks like the game actually works when I hit "Single Player" so I don't know if maybe I just got lucky or if something happens when you play the game twice successively.  Either way, when it did work, it ran flawlessly.

**Comment by kisak-valve** [comment]
Alien Swarm: Reactive Drop (563560) the text for the ingame tips are blurry and DS4 controller doesn't work

Issue transferred from https://github.com/ValveSoftware/Proton/issues/1049.
@daasuharo posted on 2018-08-31T03:20:04:

Distro: Ubuntu 18.04.1
GPU: Intel Iris Pro 580
Resolution: 1280x720
RAM: 32GB
Controller: DS4
Proton 3.7-5 Beta

https://gist.github.com/daasuharo/cb605b8cd50227982a9bfa4dfcb1f635

**Comment by parkerlreed** [comment]
Same as the other issue: training works on 4.2. Only hangs for about 5 seconds.

**Comment by kisak-valve** [comment]
Problems with chat/console

Issue transferred from https://github.com/ValveSoftware/Proton/issues/3758.
@yiitbrak posted on 2020-04-13T08:57:35:

# Compatibility Report
- Name of the game with compatibility issues: Alien Swarm:Reactive Drop
- Steam AppID of the game: 563560

## System Information
- GPU: Nvidia GT 710 
- Driver/LLVM version: nvidia 435.21 
- Kernel version: 5.3.0-46-generic 
- Link to full system information report as [Gist](https://gist.github.com/): https://gist.github.com/yiitbrak/b874b5b77c8372ee984cae5e4d588f43
- Proton version: 5.0-5

## I confirm:
- [ ] that I have checked whether there are updates for my system available.



## Symptoms 
-All kinds of text comes to chat box disappears from history between 5-10 seconds
-The font that console uses is so hard to read, mostly un visible and some letters are inside anothers

## Reproduction
-For chat  bug just write something to chat
-For console bug open console

**Comment by nididak** [comment]
I solved unredable console font bug with a mod called "Better console" from Steam workshop, chat text  bug still stays

**Comment by Wedge009** [comment]
I didn't find any issues during a quick test against Proton 7.0-1. (Previously played against at least Proton 6.3)

**Comment by BenLubar** [comment]
> ## Symptoms
> -All kinds of text comes to chat box disappears from history between 5-10 seconds

This is the result of `rd_chatwipe 1`, not Proton-specific. Set `rd_chatwipe 0` to prevent this.

**Comment by sjas** [comment]
have an ubuntu based linux installation, "alien swarm: reactive drop" still does steam cloud sync even when it is explicitly disabled in all three places and steam was restarted: (or even reinstalled afterwards, tried that as well)

- ingame in asw:rd
- for asw:rd in steam library
- globally in steam settings

tried turning off because cloud keeps on overwriting local settings, result is:

- game settings are overwritten every time a map is entered or you drop into lobby
- gear loadout is resetted as well (ofc), so you end up with wrong guns in the lobby
- i cant seem to change the gear/settings defaults that are saved in the steam cloud storage either
- even when writeproofing config files and i.e. forcing a cloud upload via steam console `cloud_sync_up 563560` (" Manually triggers after-game cloud sync for the given app.")

several complete steam reinstalls+game didnt seem to help me this time, that did work once in the past somehow with enabling/disabling cloud right away years ago

how can i get this solved short of getting a new steam account?
completely purging the game from my account changes nothing as well.

tested proton versions were 5,6,7,experimental, game worked otherwise with proper cloudsync and autoexec.cfg being read properly. `rd_revert_convars 0` or `1` doesnt change anything either.

**Comment by BenLubar** [comment]
@sjas does a symlink help? saw this ProtonDB entry that looks related: https://www.protondb.com/app/563560#-TH7Tafjj

**Comment by sjas** [comment]
@BenLubar symlink wasnt needed for me.

so far, i still cannot turn cloud sync of, at all, even when disabling it everywhere, but i seem to have forced the cloud to accept a new file.

the resets came from `rd_revert_convars 1` in the userdata folder config that came from the cloud time and time again. (was neither in the regular `config.cfg` nor my `autoexec.cfg`)

fixed the proper cloud config seemingly by these steps, which i tried with proton v5/6/7 because these were all the versions ive used in the past: (each time i set asw:rd to a fixed proton version, on the first start it asked wether the config should be cloudsynced, i chose yes everytime)

- enabled general steam cloudsync in steam settings
- enabled asw:rd cloudsync in steam library
- set fixed proton version for asw:rd in steam library
- started asw:rd
- enable ingame cloudsync when prompted
- while game is running, copying the desired config.cfg over to my userdata folder: ./steam/userdata/USER_ID/563560/remote/reactivedrop/cfg/config.cfg , and writeprotecting it
- afterwards exit game
- start and exit game again
- repeat for all proton versions
- afterwards removed the write protection again and it kept working

what i seemingly found out:
- autoexec gets executed only once upon game start? (not 1000% sure about that but thats how it looked like)
- on game start none the ondisk config.cfg's (reactivedrop/cfg one not and also not the userdata one) seem to change from the cloud even if it has different contents
- when ending the game, reactivedrop/cfg/config.cfg seems to be copied over to the userdata folder
- this contents also end up in the cloud then

you can make an educated guess on what config is in the cloud from the file sizes:
- ./steam/steamapps/common/reactivedrop/cfg/config.cfg filesize
- ./steam/userdata/USER_ID/563560/remote/reactivedrop/cfg/config.cfg filesize
- ./steam/userdata/46754621/563560/remotecache.vdf mentions the filesize of the cloud file, its a plaintext file

problem solved even though im not 100% sure why, but im definitely not complaining...

**Comment by BenLubar** [comment]
I have a report that today's update (which bumped the Steam API version from 1.38a to 1.53a) requires a newer version of Proton to run successfully. (4.2 -> 6.3)

**Comment by BenLubar** [comment]
Can Proton for Alien Swarm: Reactive Drop be made "official"? Is this something I can do in the Steamworks admin panel or something Valve has to do?

The game runs great on the Steam Deck (according to the compatibility reviews; I don't have mine yet) and has a platinum rating on ProtonDB.

---

italic text in Source Engine closed captions is cut off

Issue transferred from https://github.com/ValveSoftware/Proton/issues/6544.
@BenLubar posted on 2023-02-17T03:10:48:

Proton 7.0-6

Tested in Alien Swarm: Reactive Drop, but it also works in any other Source 1 game that has italicized closed captions.

Repro steps:

1. (On the beta branch of appid 563560)
2. Enable full closed captions in audio settings.
3. Start the first level (Jacob's Rest: Landing Bay)
4. Walk through the level until you reach a locked door. (No enemies spawn before this point on Normal difficulty.) Punch the door by pressing right mouse button or B on a controller.
5. The caption *[Door hit]* has every letter missing pieces on Proton, which is not the case on Windows.

Alternative repro steps:

In any Source 1 game, trigger a caption that has `<I>` in it.

**Comment by kisak-valve** [comment]
Hello @BenLubar, it's possible that the kerning issue you've described is related to https://github.com/ValveSoftware/steam-for-linux/issues/3057 as well.

**Comment by BenLubar** [comment]
Windows:
![](https://steamuserimages-a.akamaihd.net/ugc/2497885570415225027/49C34EF29708C7FF88DACDD0294745E97CD85406/)

Proton:
![](https://steamuserimages-a.akamaihd.net/ugc/2497885570414929793/C7A671ECE1810EE0E1B30E1FB7FE52DAECB09B80/)

It looks like all of the characters are pushed slightly to the right, causing them to both be cut off and spaced out more than they should be.

**Comment by BenLubar** [comment]
https://github.com/ValveSoftware/wine/commit/cd0172ffb83cb7e92d8884117725ae4ef1bedfd8

Proton Experimental hugely improves text legibility (it looks like it does on Windows now for non-italicized text, and there's not a lot of italicized text).

I've sent a request to get the game re-Steam-Deck-verified so the default Proton version can be changed.

**Comment by colleirose** [comment]
The entire game does not start at all for me.

# Compatibility Report
- Name of the game with compatibility issues: Alien Swarm: Reactive Drop crashes
- Steam AppID of the game: 563560

## System Information
- GPU: NVIDIA GeForce RTX 3050 Ti Mobile
- Video driver version: I don't remember if I installed the proprietary or FOSS drivers but I just ran `sudo pacman -Syu` so it's the latest driver
- Kernel version: 6.12.4-zen1-1-zen
- Link to full system information report as [Gist](https://gist.github.com/): I don't know how to get a full system information report, but here's hyfetch output: https://gist.github.com/colleirose/72609eabd5e699edee8eeafd01112b9a
- Proton version:  9.0

## Symptoms 
Nothing happens when started in steam (says you're playing, nothing shows up, says you're not playing after a few seconds).

Attempting to start manually in console gets this error:
```
Unhandled exception: page fault on read access to 0x00000000 in wow64 32-bit code (0x00000000).
Register dump:
 CS:0023 SS:002b DS:002b ES:002b FS:0063 GS:006b
 EIP:00000000 ESP:00507664 EBP:0050ff04 EFLAGS:00010202(  R- --  I   - - - )
 EAX:00000004 EBX:ffffffff ECX:00000000 EDX:00000000
 ESI:0a152fb0 EDI:1000d010
Stack dump:
0x00507664:  0a05e72e 0a0e3fd8 00000000 0050773c
0x00507674:  00000002 1000d010 1000d511 1000d010
0x00507684:  1002a568 0050773c 1000d98f 0050773c
0x00507694:  1000d9ee 005076cc 1000d9fe 1002a568
0x005076a4:  00b51b30 10004e15 00353de2 00000000
0x005076b4:  00400000 10010821 1001083f 10000000
Backtrace:
=>0 0x00000000 (0x0050ff04)
  1 0x00401503 in reactivedrop (+0x1503) (0x0050ff50)
  2 0x7becea18 in kernel32 (+0xea18) (0x0050ff68)
  3 0x7bf4cb13 in ntdll (+0xcb13) (0x0050ff80)
  4 0x7bf7f5c2 in ntdll (+0x3f5c2) (0x0050ffec)
0x00000000: -- no code accessible --
Modules:
Module  Address                 Debug info      Name (98 modules)
PE        400000-  406000       Export          reactivedrop
PE        b50000-  bcf000       Deferred        tier0
PE        bd0000-  c3b000       Deferred        vstdlib
PE       8480000- 84da000       Deferred        filesystem_stdio
PE       9570000- 9c62000       Deferred        engine
PE       a020000- a046000       Deferred        inputsystem
PE       a050000- a194000       Deferred        materialsystem
PE       b1c0000- b207000       Deferred        datacache
PE       b210000- b662000       Deferred        studiorender
PE       b670000- b785000       Deferred        vphysics
PE       b790000- b84b000       Deferred        vscript
PE       b850000- b874000       Deferred        valve_avi
PE       b880000- b99f000       Deferred        vguimatsurface
PE       b9a0000- b9fc000       Deferred        vgui2
PE       bab0000- bc5f000       Deferred        shaderapidx9
PE       bc60000- be0f000       Deferred        shaderapidx9
PE       be10000- be35000       Deferred        localize
PE      10000000-10034000       Deferred        launcher
PE      18000000-18033000       Deferred        binkw32
PE-Wine 79ff0000-7a127000       Deferred        opengl32
PE-Wine 7a140000-7a2f0000       Deferred        wined3d
PE-Wine 7a300000-7a336000       Deferred        d3d9
PE-Wine 7a350000-7a462000       Deferred        oleaut32
PE-Wine 7a480000-7a5d2000       Deferred        comctl32
PE-Wine 7a5f0000-7a613000       Deferred        msvfw32
PE-Wine 7a630000-7a674000       Deferred        avifil32
PE-Wine 7a690000-7a6ae000       Deferred        mpr
PE-Wine 7a6c0000-7a746000       Deferred        wininet
PE-Wine 7a760000-7a781000       Deferred        msacm32
PE-Wine 7a7a0000-7a864000       Deferred        winmm
PE-Wine 7a880000-7a8b6000       Deferred        uxtheme
PE-Wine 7a8d0000-7a8e6000       Deferred        winex11
PE      7a900000-7a943000       Deferred        steam_api
PE-Wine 7a960000-7a973000       Deferred        coml2
PE-Wine 7a990000-7aa15000       Deferred        rpcrt4
PE-Wine 7aa30000-7aa7d000       Deferred        combase
PE-Wine 7aa90000-7aba4000       Deferred        ole32
PE-Wine 7abc0000-7abd6000       Deferred        shcore
PE-Wine 7abf0000-7ac39000       Deferred        shlwapi
PE-Wine 7ac50000-7b57a000       Deferred        shell32
PE-Wine 7b590000-7b5b8000       Deferred        ws2_32
PE-Wine 7b5d0000-7b5db000       Deferred        nsi
PE-Wine 7b5f0000-7b604000       Deferred        dnsapi
PE-Wine 7b620000-7b644000       Deferred        iphlpapi
PE-Wine 7b660000-7b66f000       Deferred        wsock32
PE-Wine 7b680000-7b69c000       Deferred        imm32
PE      7b6b0000-7b6c5000       Deferred        vcruntime140
PE-Wine 7b6e0000-7b710000       Deferred        win32u
PE-Wine 7b720000-7b7a7000       Deferred        gdi32
PE      7b7c0000-7b8a1000       Deferred        ucrtbase
PE-Wine 7b8c0000-7b8e8000       Deferred        sechost
PE-Wine 7b900000-7b9b2000       Deferred        msvcrt
PE-Wine 7b9d0000-7ba12000       Deferred        advapi32
PE-Wine 7ba30000-7bbe2000       Deferred        user32
PE-Wine 7bc00000-7bea3000       Deferred        kernelbase
PE-Wine 7bec0000-7bf25000       COFF            kernel32
PE-Wine 7bf40000-7bfee000       COFF            ntdll
ELF     eade2000-eae80000       Deferred        opengl32.so
ELF     eae80000-eae8c000       Deferred        libxcursor.so.1
ELF     eae8c000-eae9f000       Deferred        libxi.so.6
ELF     eae9f000-eaea4000       Deferred        libxcomposite.so.1
ELF     eaea4000-eaeab000       Deferred        libxfixes.so.3
ELF     eaeab000-eaebb000       Deferred        libxrandr.so.2
ELF     eaebb000-eaec7000       Deferred        libxrender.so.1
ELF     eaec7000-eaece000       Deferred        libxxf86vm.so.1
ELF     eaece000-eaefb000       Deferred        libxcb.so.1
ELF     eaefb000-eb036000       Deferred        libx11.so.6
ELF     eb060000-eb0f1000       Deferred        winex11.so
ELF     eb0f1000-eb103000       Deferred        libresolv.so.2
ELF     eb105000-eb10a000       Deferred        libxinerama.so.1
ELF     eb10a000-eb111000       Deferred        libxdmcp.so.6
ELF     eb111000-eb127000       Deferred        libxext.so.6
ELF     eb127000-eb12d000       Deferred        ws2_32.so
ELF     eb14b000-eb175000       Deferred        libexpat.so.1
ELF     eb175000-eb1c8000       Deferred        libfontconfig.so.1
ELF     eb1c8000-eb261000       Deferred        libpcre2-8.so.0
ELF     eb261000-eb284000       Deferred        libbrotlicommon.so.1
ELF     eb284000-eb3d1000       Deferred        libglib-2.0.so.0
ELF     eb3d1000-eb406000       Deferred        libgcc_s.so.1
ELF     eb406000-eb415000       Deferred        libbrotlidec.so.1
ELF     eb415000-eb539000       Deferred        libharfbuzz.so.0
ELF     eb539000-eb573000       Deferred        libpng16.so.16
ELF     eb573000-eb586000       Deferred        libbz2.so.1.0
ELF     eb586000-eb647000       Deferred        libfreetype.so.6
ELF     eb647000-eb71f000       Deferred        libm.so.6
ELF     eb71f000-eb724000       Deferred        libxau.so.6
ELF     eb749000-eb8a7000       Deferred        win32u.so
ELF     eb8af000-eb8b4000       Deferred        librt.so.1
ELF     eb8b4000-eb99a000       Export          ntdll.so
ELF     eb99c000-ebbd1000       Deferred        libc.so.6
ELF     ebbd1000-ebbd6000       Deferred        libdl.so.2
ELF     ebbd6000-ebbdb000       Deferred        libpthread.so.0
ELF     ebbe0000-ebbe5000       Deferred        dnsapi.so
ELF     ebbe8000-ebc00000       Deferred        libz.so.1
ELF     ebc07000-ebc3d000       Deferred        ld-linux.so.2
ELF     ebc3d000-ebc42000       Deferred        <wine-loader>
PE-Wine ff990000-ffb40000       Deferred        wined3d
PE-Wine ffb40000-ffb76000       Deferred        d3d9
Threads:
process  tid      prio    name (all IDs are in hex)
00000020 (D) Z:\home\c\.local\share\Steam\steamapps\common\Alien Swarm Reactive Drop\reactivedrop.exe
	00000024    0 <== 
	0000013c    0     
	00000140    0     
	00000144    0     
	00000158    0     
00000038 services.exe
	0000003c    0     
	00000040    0     wine_rpcrt4_server
	0000004c    0     wine_rpcrt4_io
	00000050    0     wine_threadpool_worker
	0000006c    0     wine_rpcrt4_io
	0000008c    0     wine_rpcrt4_io
	000000c0    0     wine_rpcrt4_io
	000000c4    0     wine_threadpool_worker
	000000e8    0     wine_threadpool_worker
	000000ec    0     wine_threadpool_waitqueue
	000000fc    0     
	00000108    0     wine_rpcrt4_io
00000044 svchost.exe
	00000048    0     
	00000054    0     
	00000058    0     wine_sechost_service
0000005c winedevice.exe
	00000060    0     
	00000070    0     
	00000074    0     wine_sechost_service
	00000078    0     
	0000007c    0     
	00000080    0     
	000000f4    0     
	000000f8    0     
00000064 explorer.exe
	00000068    0     
	000000b4    0     
	000000b8    0     wine_rpcrt4_server
00000084 winedevice.exe
	00000088    0     
	00000090    0     
	00000094    0     wine_sechost_service
	00000098    0     
	0000009c    0     
	000000a8    0     
	000000f0    0     
000000ac plugplay.exe
	000000b0    0     
	000000c8    0     
	000000cc    0     wine_sechost_service
	000000d0    0     wine_rpcrt4_server
00000100 rpcss.exe
	00000104    0     
	0000010c    0     
	00000110    0     wine_sechost_service
	00000114    0     wine_rpcrt4_server
	00000118    0     wine_rpcrt4_server
	0000011c    0     wine_rpcrt4_io
	00000120    0     wine_threadpool_worker
00000124 tabtip.exe
	00000128    0     
	00000130    0     wine_rpcrt4_server
	00000134    0     
00000150 conhost.exe
	00000154    0     
System information:
    Wine build: wine-9.0
    Platform: x86_64 (guest: i386)
    Version: Windows 10
    Host system: Linux
    Host version: 6.12.4-zen1-1-zen
``

The game ironically starts fine in vanilla Wine, only reason it doesn't play is because of the Steam DRM.

**Comment by Wedge009** [comment]
Hmm, did it ever work for you? I just gave it a quick test against Proton 9.0-4, no apparent issue, but my system is rather different to yours - Ubuntu-based, AMD GPU, etc.

It might be useful to provide a Proton log to help others to help you here.

**Comment by colleirose** [comment]
No, it has not worked for me.

[steam-563560.log](https://github.com/user-attachments/files/18235698/steam-563560.log)

**Comment by kisak-valve** [comment]
Hello @colleirose, please copy the contents of Steam Runtime Diagnostics from Steam (`Steam` -> `Help` -> `Steam Runtime Diagnostics`) and put it in a [gist](https://gist.github.com/), then include a link to the gist in this issue report.

DXVK failing to initialize usually hints that a Vulkan render path isn't usable.

**Comment by colleirose** [comment]
@kisak-valve I have added a link to the diagnostic report

**Comment by kisak-valve** [comment]
Thanks, https://gist.github.com/colleirose/9a970bf59b9e2c08ac02b00c60a929d0#file-gistfile1-txt-L268-L281 is the section to focus on. Vulkan is broken in general on your system, but it's not immediately clear how it broke.

**Comment by Zelly9191** [comment]
This issue with italicized text is affecting most source mods as well. And it doesn't seem like proton experimental has changed that at least the current version. I've tried several versions of proton with no success. Not sure if valve knows about this but I would assume they do as they entirely disabled italicized text in the 20th anniversary update of Half-Life 2. 

https://github.com/ValveSoftware/Source-1-Games/issues/6580

It's great that it's not an issue in Half-Life 2 atm, but it's really more of a Band-Aid fix as all mods and source games still have this problem.

To clarify, I'm running into this issue on my og steam deck
